src/sources/aws_ecs_metrics/parser.rs

```diff
@@ -258,65 +258,102 @@ fn cpu_metrics(
     namespace: &Option<String>,
     tags: &BTreeMap<String, String>,
 ) -> Vec<Metric> {
-    let mut metrics = vec![gauge(
-        "cpu",
-        "online_cpus",
-        namespace.clone(),
-        timestamp,
-        cpu.online_cpus as f64,
-        tags.clone(),
-    )];
+    let mut metrics = vec![];
 
-    metrics.extend(
-        vec![
-            ("usage_system_jiffies_total", cpu.system_cpu_usage),
-            (
-                "usage_usermode_jiffies_total",
-                cpu.cpu_usage.usage_in_usermode,
-            ),
-            (
-                "usage_kernelmode_jiffies_total",
-                cpu.cpu_usage.usage_in_kernelmode,
-            ),
-            ("usage_total_jiffies_total", cpu.cpu_usage.total_usage),
-            ("throttling_periods_total", cpu.throttling_data.periods),
-            (
-                "throttled_periods_total",
-                cpu.throttling_data.throttled_periods,
-            ),
-            (
-                "throttled_time_seconds_total",
-                cpu.throttling_data.throttled_time / 1_000_000_000.0,
-            ),
-        ]
-        .iter()
-        .map(|(name, value)| {
-            counter(
-                "cpu",
-                name,
-                namespace.clone(),
-                timestamp,
-                *value,
-                tags.clone(),
-            )
-        }),
-    );
+    if let Some(online_cpus) = cpu.online_cpus {
+        metrics.extend(vec![gauge(
```

Comment:
I think you can just use `metrics.push` rather than `extend` in the cases you are only adding one metric at a time.

Reply:
doh! That's an obvious one, my bad.